openai/table_openai_completion.go

```diff
@@ -137,28 +129,25 @@ func listCompletion(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateD
 		}
 	}
 
-	// Both are valid, but the order of precedence is:
-	// 1. prompt = "my prompt"
-	// 2. settings = '{"prompt": "my prompt"}'
-	if d.EqualsQuals["prompt"] != nil {
-		cr.Prompt = d.EqualsQualString("prompt")
-	}
-
-	if cr.Prompt == "" {
+	if d.EqualsQuals["prompt"] == nil {
 		// No prompt, so return zero rows
 		return nil, nil
 	}
 
-	plugin.Logger(ctx).Debug("openai_completion.listCompletion", "prompt", cr)
-	resp, err := conn.CreateCompletion(ctx, cr)
+	prompt := d.EqualsQualString("prompt")
+	plugin.Logger(ctx).Debug("openai_completion.listCompletion", "prompt", prompt)
+
+	resp, err := conn.CreateChatCompletion(context.Background(), cr)
+
 	if err != nil {
 		plugin.Logger(ctx).Error("openai_completion.listCompletion", "prompt", cr, "completion_error", err)
 		return nil, err
 	}
 	plugin.Logger(ctx).Debug("openai_completion.listCompletion", "completion_response", resp)
 
 	for _, i := range resp.Choices {
-		row := CompletionRow{i, cr.Prompt}
+		row := CompletionRow{i.Message.Content, prompt}
+		plugin.Logger(ctx).Debug("openai_completion.listCompletion", "row", row)
```

Comment:
I kept `plugin.Logger(ctx).Debug("openai_completion.listCompletion", "row", row)` if that's ok, to make debugging simpler.

Added the context related line.

I imported  "github.com/jinzhu/copier"  to make LogProbs deepcopy easier and created new CompletionRow to return

Reply:
@nicocesar, Including debug statements in the code is beneficial, though they can sometimes clutter the log file. It might be more advantageous to keep the log statements focused on essential information. Would you consider removing or adjusting them to reduce noise?